### PR TITLE
Unified KeyTuple into a new type called FourTuple. Also introduced Fo…

### PR DESCRIPTION
**Pull request by DanielLavie**
### What does this PR do?

This PR combines http.KeyTuple and kafka.KeyTuple into a new type called ConnectionKey and introduces ConnectionKeysFromConn, which is a merged version of HTTPKeyTuplesFromConn and KafkaKeyTuplesFromConn.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/types/four_tuple.go
Comment (guyarb) [inline pkg/network/types/four_tuple.go:1]: i'm not a big fan of the naming
let's go with something that represents it like `ConnectionKey` or something that represents the meaning, and not the implementation

saying, tomorrow we might add another indicator (maybe pid? maybe tcp/udp?) and then fourtuple won't be a good enough name.

File: pkg/network/encoding/http.go [hunk 1]
```diff
@@ -11,24 +11,24 @@ import (
...
+	staticTags     map[types.FourTuple]uint64
+	dynamicTagsSet map[types.FourTuple]map[string]struct{}
 
 	orphanEntries int
 }
 
 // aggregationWrapper is meant to handle collision scenarios where multiple
 // `ConnectionStats` objects may claim the same `HTTPAggregations` object because
-// they generate the same http.KeyTuple
+// they generate the same four tuple
```
Comment (guyarb) [inline pkg/network/encoding/http.go:28]: same same

**Review by guyarb — CHANGES_REQUESTED** [review]

File: pkg/network/types/four_tuple.go
Comment (DanielLavie) [reply pkg/network/types/four_tuple.go:1]: Changed to ConnectionKey

File: pkg/network/encoding/http.go (hunk 1 above)
Comment (DanielLavie) [reply pkg/network/encoding/http.go]: Changed to connection key

**Review by guyarb — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by heyronhay — APPROVED** [review]

**Review by derekwbrown — APPROVED** [review]

**Pull request merged**